src/storage/mod.rs

```diff
@@ -317,15 +318,18 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                         .get(CMD)
                         .observe(command_duration.elapsed_secs());
 
-                    result
+                    Ok((result, statistics, perf_statistics.delta()))
```

Comment:
`(result, statistics, perf_statistics.delta())`

Reply:
if we do it, we can't use `?` at above.

Reply:
> if we do it, we can't use `?` at above.

Get your point. Seems I misled you. Could you revert these changes? Sorry about that.

Reply:
Ok, don't worry, I think It's excellent that the function's return type wrap by `Optional` or `Result` in Rust.